### ISSUE TITLE
Self-contained .NET Core apps can't use .NET Core COM components

### ISSUE THREAD

**Issue by vitek-karas** [dotnet/core-setup#7610]
#### Observed behavior
Self-contained apps can't natively host other .NET Core components.
Natively hosted components are:
* COM objects which are implemented with .NET Core
* IJW assemblies in some cases
* WinRT objects which are implemented with .NET Core
* explicit native hosting of .NET Core components using `hostfxr` APIs like `hostfxr_initialize_for_runtime_config`.

#### Repro
Clone the dotnet/samples repo.
Go to `core/extensions/COMServer` sample.
`dotnet build`
Register the server from elevated command prompt as directed by the build output.
Go to `COMClient` folder.
`dotnet publish --self-contained -r win-x64` (it's likely you will need to delete the `NuGet.config` in the root of the sample for this to work).
`bin\Debug\netcoreapp3.0\win-x64\publish\COMClient.exe`

It should correctly print out the PI number.
Instead it fails like this:
```
The specified framework 'Microsoft.NETCore.App' is not present in the previously loaded runtime.
Unhandled exception. System.Runtime.InteropServices.COMException (0x800080A5): Retrieving the COM class factory for component with CLSID {DB1797F5-7198-4411-8563-D05F4E904956} failed due to the following error: 800080a5 0x800080A5.
   at COMClient.Program.Main(String[] args) in F:\dotnet\samples\core\extensions\COMServerDemo\COMClient\Program.cs:line 10
```

#### Root cause
The root cause of the problem is validation done by `hostfxr` when dynamically loading components via the native hosting APIs. The component (in this case the COM server) is always framework dependent (we don't support self-contained components), and as such it includes `.runtimeconfig.json` which defines the required framework dependencies (In this case just the `Microsoft.NETCore.App`). `hostfxr` validates that the component framework dependencies can be fulfilled by the currently running .NET Core runtime in the process. It does this by comparing the declared framework dependencies to the frameworks loaded into the process. Self-contained apps don't have any frameworks, the app and the framework are merged into one entity and we don't remember what frameworks and which versions were used. The validation then fails because it thinks the app can't fulfill the framework requirements of the component.

**Comment by vitek-karas** [comment]
#7628 implements a fix for 3.0. The proper long term fix will require SDK changes and additional information in `.runtimeconfig.json`.

**Comment by jeffschwMSFT** [comment]
@vitek-karas would you like to keep this issue open to track the long term?  My preference would be to close this (as a fix for the short term) and open a new issue to track future work.

**Comment by vitek-karas** [comment]
The 3.0 problem has been resolved by #7628. The followup changes to start validation framework references again are tracked by #7732.
